src/actions/CommandAction.ts

```diff
@@ -95,31 +86,42 @@ export default class CommandAction implements Action {
 
   async run(
     [targets]: [TypedSelection[]],
-    args: CommandArguments = {}
+    options: CommandOptions = {}
   ): Promise<ActionReturnValue> {
-    const actualArgs = Object.assign({}, defaultArguments, this.args, args);
+    const partialOptions = Object.assign(
+      {},
+      defaultOptions,
+      this.options,
+      options
+    );
+
+    if (partialOptions.command == null) {
+      throw Error("Command id must be specified");
+    }
+
+    const actualOptions = partialOptions as Required<CommandOptions>;
```

Comment:
Required was a smart move. Didn't think about that one.

Reply:
Yeah I love all those type transformations that typescript has

Reply:
That type system in typescript is way powerful.